### PR TITLE
FEAT: add CNVKIT + base skeleton

### LINKED ISSUE #3: panel of normals creation for cnvkit
### Description of feature

Given a number of normal samples a panel of normals file should be generated which can be used in cnvkit.

### PR DESCRIPTION
**Pull request by maxulysse**
Close #3

### CODE REVIEW

File: assets/schema_input.json [hunk 1]
```diff
@@ -12,25 +12,67 @@
...
+                "anyOf": [
+                    {
+                        "type": "string",
+                        "pattern": "^\\S+\\.bam$"
+                    },
+                    {
+                        "type": "string",
+                        "maxLength": 0
+                    }
+                ],
```
Comment (nvnieuwk) [inline assets/schema_input.json:26]:
```suggestion
                "type": "string",
                "pattern": "^\\S+\\.bam$",
```

You don't need to specify empty fields in the schema, `nf-validation` automatically skips validation of empty fields, if they are not required :). And I think `anyOf` doesn't work for the conversion to a channel (I've not tested it though).

File: nextflow_schema.json [hunk 2]
```diff
@@ -13,14 +13,23 @@
...
+                            "type": "string",
+                            "format": "file-path",
+                            "exists": true,
+                            "mimetype": "text/csv",
+                            "pattern": "^\\S+\\.csv$"
+                        },
+                        {
+                            "type": "string",
+                            "maxLength": 0
+                        }
```
Comment (nvnieuwk) [inline nextflow_schema.json:31]: Same here :)

File: workflows/createpanelrefs.nf [hunk 3]
```diff
@@ -63,32 +96,16 @@ workflow CREATEPANELREFS {
...
-
-    //
-    // MODULE: Run FastQC
-    //
-    FASTQC (
-        INPUT_CHECK.out.reads
-    )
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
+    if (params.tools && params.tools.split(',').contains('cnvkit')) {
+        CNVKIT_BATCH ( ch_input.bam.map{meta, bam -> [ meta, [], bam ]}, ch_fasta, [], [], [], true )
```
Comment (nvnieuwk) [inline workflows/createpanelrefs.nf:100]: Why don't you also convert the CRAM files to BAM and add them here?

**Review by nvnieuwk — COMMENTED** [review]
Some pointers, but overall LGTM :)

File: assets/schema_input.json (hunk 1 above)
Comment (maxulysse) [reply assets/schema_input.json:26]: I copy paste code from another PR, if it's ok with you, I'd rather fix this part in a dedicated PR

File: nextflow_schema.json (hunk 2 above)
Comment (maxulysse) [reply nextflow_schema.json:31]: same here ;-)

File: assets/schema_input.json (hunk 1 above)
Comment (nvnieuwk) [reply assets/schema_input.json:26]: Yeah that's okay! :)

File: workflows/createpanelrefs.nf (hunk 3 above)
Comment (maxulysse) [reply workflows/createpanelrefs.nf:100]: Let's do that in a separate PR.
I'd love to have the basic skeleton included that way I can add more tools and work on more features separately

Comment (nvnieuwk) [reply workflows/createpanelrefs.nf:100]: Allright fine by me :)

**Review by nvnieuwk — APPROVED** [review]

**Pull request merged**